server/src/main/kotlin/suwayomi/tachidesk/opds/impl/OpdsFeedBuilder.kt

```diff
@@ -0,0 +1,1016 @@
+                    sortColumn,
+                    currentSortOrder,
+                )
+                entries.addAll(chapterEntries.map { chapter -> createChapterListEntry(chapter, mangaDetails, baseUrl, false, locale) })
+            }
+        return OpdsXmlUtil.serializeFeedToString(builder.build())
+    }
+
+    suspend fun getChapterMetadataFeed(
+        mangaId: Int,
+        chapterSourceOrder: Int,
+        baseUrl: String,
+        locale: Locale,
+    ): String {
+        val mangaDetails =
+            MangaRepository.getMangaDetails(mangaId)
+                ?: return buildNotFoundFeed(
+                    baseUrl,
+                    "manga/$mangaId/chapter/$chapterSourceOrder/metadata",
+                    MR.strings.opds_error_manga_not_found.localized(locale, mangaId.toString()),
+                    locale,
+                )
+
+        val chapterMetadata =
+            ChapterRepository.getChapterDetailsForMetadataFeed(mangaId, chapterSourceOrder)
+                ?: return buildNotFoundFeed(
+                    baseUrl,
+                    "manga/$mangaId/chapter/$chapterSourceOrder/metadata",
+                    MR.strings.opds_error_chapter_not_found.localized(locale, chapterSourceOrder.toString()),
+                    locale,
+                )
+
+        val builder =
+            FeedBuilderInternal(
+                baseUrl = baseUrl,
+                idPath = "manga/$mangaId/chapter/${chapterMetadata.sourceOrder}/metadata",
+                title = MR.strings.opds_feeds_chapter_details.localized(locale, mangaDetails.title, chapterMetadata.name),
+                locale = locale,
+                feedType = OpdsConstants.TYPE_ATOM_XML_FEED_ACQUISITION,
+                pageNum = null, // Metadata feed is single entry, not paginated
+            ).apply {
+                totalResults = 1
+                icon = mangaDetails.thumbnailUrl?.let { proxyThumbnailUrl(mangaDetails.id) }
+                icon?.let {
+                    links.add(OpdsLinkXml(OpdsConstants.LINK_REL_IMAGE, it, OpdsConstants.TYPE_IMAGE_JPEG))
+                    links.add(OpdsLinkXml(OpdsConstants.LINK_REL_IMAGE_THUMBNAIL, it, OpdsConstants.TYPE_IMAGE_JPEG))
+                }
+                entries.add(createChapterMetadataEntry(chapterMetadata, mangaDetails, locale))
+            }
+        return OpdsXmlUtil.serializeFeedToString(builder.build())
+    }
+
+    fun getSourceFeed(
+        sourceId: Long,
+        baseUrl: String,
+        pageNum: Int,
+        locale: Locale,
+    ): String {
+        val (mangaEntries, total) = MangaRepository.getMangaBySource(sourceId, pageNum)
+        val sourceNavEntry = NavigationRepository.getSources(1).first.find { it.id == sourceId }
+        val feedTitle =
+            MR.strings.opds_feeds_source_specific_title.localized(
+                locale,
+                sourceNavEntry?.name ?: MR.strings.opds_feeds_source_specific_title.localized(locale, sourceId.toString()),
```

Comment:
If you are going to pass a numer, set the string xml to take a number instead of a string. Make sure to update both the translated files

Here is the relavent documentation:
https://developer.android.com/reference/kotlin/java/util/Formatter.html#conversions